core/src/impl/Kokkos_BitOps.hpp

```diff
@@ -82,13 +82,16 @@ inline int int_log2_device(unsigned i) {
 
 KOKKOS_IMPL_HOST_FUNCTION
 inline int int_log2_host(unsigned i) {
-  constexpr int shift = sizeof(unsigned) * CHAR_BIT - 1;
+// duplicating shift to avoid unused warning in else branch
```

Comment:
Could you stick a `FIXME` on this? Once we flip to C++17, we can mark it `[[maybe_unused]]`

Reply:
I really would like to avoid restarting the CI yet again, we need to move this to finally get the release process started.

Reply:
I think we can replace pretty much all `(void) bla` with `[[maybe_unused]] bla` and don't need an extra comment here.